static/js/stream_edit.js

```diff
@@ -211,9 +211,18 @@ function submit_add_subscriber_form(e) {
         return;
     }
 
-    const user_ids = user_pill.get_user_ids(exports.pill_widget);
     const stream_subscription_info_elem = $(".stream_subscription_info").expectOne();
+    let user_ids = user_pill.get_user_ids(exports.pill_widget);
+    user_ids = user_ids.concat(stream_pill.get_user_ids(exports.pill_widget));
+    user_ids = Array.from(new Set(user_ids));
 
+    if (user_ids.includes(page_params.user_id) && sub.subscribed) {
+        // We don't want to send a request to subscribe ourselves
+        // if we are already subscribed to this stream. This
+        // case occurs when creating user pills from a stream.
+        const index = user_ids.indexOf(page_params.user_id);
+        user_ids.splice(index, 1);
+    }
```

Comment:
Couldn't this be done much more simply while we still have a `Set` by just removing the current user ID, rather than doing this awkward array splicing?